src/main/host/thread_preload.c

```diff
@@ -126,21 +140,6 @@ static pid_t _threadpreload_fork_exec(ThreadPreload* thread, const char* file, c
         case 0: {
             // child
 
-            // Ensure that the child process exits when Shadow does.  Shadow
-            // ought to have already tried to terminate the child via SIGTERM
-            // before shutting down (though see
-            // https://github.com/shadow/shadow/issues/903), so now we jump all
-            // the way to SIGKILL.
-            if (prctl(PR_SET_PDEATHSIG, SIGKILL) < 0) {
-                error("prctl: %s", g_strerror(errno));
-                return -1;
-            }
-            // Validate that Shadow is still alive (didn't die in between forking and calling
-            // prctl).
-            if (getppid() != shadow_pid) {
-                error("parent (shadow) exited");
-                return -1;
-            }
             int rc = execvpe(file, argv, envp);
```

Comment:
I think we want a comment somewhere around here about how the code is designed so that the child process does not execute any code after the `vfork` and before calling `exec`, to guard against future programmers changing this code from accidentally causing undefined behavior.

Reply:
Fixed